src/lib/containers/UpsetPlot.tsx

```diff
@@ -123,7 +123,11 @@ const UpsetPlot: React.FunctionComponent<UpsetPlotProps> = ({
       } catch (err) {
         if (!isCancelled) {
           setIsLoading(false)
-          setError(err.toString())
+          let errorMessage = err['reason']
+          if (!errorMessage) {
+            errorMessage = err.toString()
+          }
```

Comment:
Just curious, is err a JSON object and you will cast it to string?

Reply:
Not sure what err is.  If it's an error that resulted from getFullQueryTableResults() async call, then it will very likely be a JSON object (that contains the "reason" key).  Is there a better approach to this (unknown runtime error handling)?

Reply:
Ah, I see. I would probably just hardcode some general error message like "Error retrieving data" if the message is user facing. At the same time I would console.log the error object for inspection.